src/libtcod/zip_c.c

```diff
@@ -165,6 +165,12 @@ void TCOD_zip_put_console(TCOD_zip_t zip, const TCOD_Console* val)
 	}
 }
 
+void TCOD_zip_put_random(TCOD_zip_t zip, const TCOD_Random *val) {
+	size_t s = sizeof(*val);
```

Comment:
Since the zip functions only accept int's the conversion to int should probably be explicit:
```suggestion
	int s = (int)sizeof(*val);
```